### PR TITLE
client/authority-discovery: Do not double encode signature

### PR DESCRIPTION
**Pull request by mxinden**
Previously, when publishing ones address onto the DHT, the signature
signing those addresses would be SCALE encoded twice.

This commit removes the second encoding and adjusts the tests
to catch future regressions.

### CODE REVIEW

File: client/authority-discovery/src/tests.rs [hunk 1]
```diff
@@ -225,18 +236,21 @@ fn new_registers_metrics() {
 }
 
 #[test]
-fn publish_ext_addresses_puts_record_on_dht() {
+fn request_addresses_of_others_triggers_dht_get_query() {
```
Comment (mxinden) [inline client/authority-discovery/src/tests.rs:239]: The commit merges `publish_ext_addresses_puts_record_on_dht` and `handle_dht_events_with_value_found_should_call_set_priority_group` to ensure the whole cycle works. It does not touch `request_addresses_of_others_triggers_dht_get_query`.

Sadly the tests' setup code is similar, thus the git diff is a bit messy.

**Comment by gavofyork** [comment]
polkadot companion needed?

**Comment by mxinden** [comment]
I am pretty certain that these failures are not related to the pull request. It does not touch any public APIs.

**Comment by bkchr** [comment]
> polkadot companion needed?

No, he did not used the latest Substrate master for creating this pr. So, Polkadot fails to compile.

**Pull request merged**